### ISSUE TITLE
Unnatural terrain/biome placement (OCD issue)

### ISSUE THREAD

**Issue by Quizer9O8** [Team-RTG/Realistic-Terrain-Generation#1047]
my OCD senses are tingling. I'm sure I've mentioned this before and probably many people too but this is bugging me out crazy. Look at those jaged edges. I've noticed that this happens especially with the Stone Beach, Beach, Desert and Savanna biome.
![the bane of ocd](https://cloud.githubusercontent.com/assets/19509275/20246848/022867c6-a9c0-11e6-88f2-8fdc65e5e7cb.png)
Ugh... It feels so unnatural compared to the vanilla biome borders which curves smoothly next to each other but irregular enough for it to look natural. I'm sure you can fix this way of biome border generation, which will be the right decision in the end. I really hope that this won't be a thing in the future. I really love using this mod myself and I just want to see it thrive in the modding community.

**Comment by srs-bsns** [comment]
1. You didn't provide any version information for RTG or any other mods you are using.
2. RTG beaches do not naturally extend that far in world height. The stone part should not be there.

Are you using GeographiCraft/ClimateControl with `B:wideBeaches` set to `true`?

**Comment by Quizer9O8** [comment]
Wide Beaches are set to false and this is happening in RTG-1.10.2-4.1.1.7 only.
I've checked it the other day with GC/CC but the same outcome happens.

**Comment by srs-bsns** [comment]
Can you provide a seed for that world?

**Comment by Quizer9O8** [comment]
Seed: -5223921567637570492
(Included with GC config)
Config GC: <details># Configuration file

"assorted parameters" {
    # Use more random generators, better behaved rivers, and larger rare climates
    B:"0.5 generation"=true

```
# Use 1.6 compatibility mode
B:1.6Compatibility=false

# place little islands near shore rather than in deep ocean
B:"Allow Derpy Islands"=false

# Biome size, exponential: 4 is regular and 6 is large biomes
I:"Biome Size"=6

# per thousand; vanilla is 10
I:"Mushroom Island Incidence"=10

# generate as if CC weren't on; for loading pre-existing worlds or just preventing chunk boundaries
B:"No Generation Changes"=false

# suppress Bop sub-biome generation
B:NoBoPSubBiomes=true

# Percentage of rivers prevented; changes cause chunk boundaries at some rivers
I:PercentRiverReduction=0

# increase smoothing steps; also shrinks unusual biomes some; changing produces occaisional chunk walls
B:"Smoother Coastlines"=true

# add testing routines and crash in suspicious circumstances
B:TestingMode=false
B:VanillaBiomesOn=true

# Generate land masses and climate zones similar to vanilla 1.7
B:VanillaLandAndClimate=false

# True for triple-width rivers
B:WiderRivers=false

# impose GeographiCraft generation on the Biomes o' Plenty world type
B:alterBoPWorlds=true

# impose GeographiCraft generation on Highlands world types
B:alterHighlandsWorlds=true

# biomes not saved on edges; more than 3 has no effect; -1 deactivates saving biomes
I:biomeRingsNotSaved=3

# Number of Chunk Biome layouts cached. Above 500 is ignored. 0 or below shuts off chunk info caching
I:cacheSize=0

# climate not saved on edges; -1 deactivates saving climates
I:climateRingsNotSaved=3

# Have Climate Control set the biomes for village generation; may be incompatible with village mods
B:controlVillageBiomes=false

# Comma-delimited list of externalBiome Names. No quotes or line returnsYou will have to reload Minecraft after changing this.Names not in the list aren't removed from the configs but they have no effect
S:externalBiomeNames=

# force small continent near origin
B:forceStartContinent=true

# max height value for allowing rivers; 10.0 allows everything; 0.75 is plains but no hills
D:maxChasm=10.0

# Maximum Number of Rescue attempts. Negative numbers mean no limit
I:rescueLimit=0

# subbiomes not saved on edges, default 3, -1 deactivates saving sub-biomes
I:subBiomeRingsNotSaved=-1

# suppress GeographiCraft generation in default, large biomes, and amplified worlds
B:suppressInStandardWorlds=false

# True for double-width beaches
B:wideBeaches=false
```

}

##########################################################################################################

# climate incidences

#--------------------------------------------------------------------------------------------------------#

# Blocks of land are randomly assigned to each zone with a frequency proportional to the incidence. Processing eliminates some extreme climates later, especially for quarter size zones. Consider doubling hot and snowy incidences for quarter size zones.

##########################################################################################################

"climate incidences" {
    # relative incidence of cool zones like forest/plains/hills/taiga/roofed forest
    I:"Cool Zone Incidence"=1

```
# relative incidence of hot zones like savanna/desert/plains/mesa
I:"Hot Zone Incidence"=1

# relative incidence of snowy zones
I:"Snowy Zone Incidence"=1

# relative incidence of warm zones like forest/plains/hills/jungle/swamp
I:"Warm Zone Incidence"=1
```

}

##########################################################################################################

# climate zone parameters

#--------------------------------------------------------------------------------------------------------#

# Full-size is similar to 1.7 defaults. Half-size creates zones similar to pre-1.7 snowy zones. Quarter-size creates fairly small zones but the hot and snowy incidences are limited

##########################################################################################################

"climate zone parameters" {
    # True freezes oceans in snowy latitudes. Only used with latitudinal climates.
    B:"Frozen Icecaps"=true

```
# half the climate zone size from vanilla, unless quartering
B:"Half Zone Size"=false

# Place mountains in chains
B:"Mountains in Mountain Chains"=true

# quarter the climate zone size from vanilla
B:"Quarter Zone Size"=false

# ignore climate zones altogether
B:"Random Biomes"=false

# Number of climate zones to shift the band from the default of the warm - to - cool transition at 0. Positive numbers shift the bands up.
I:bandedClimateOffset=0

# Width of banded climates (climate depends on latitude). 0 or less for normal rules. Width is in terms of climate zones, whatever they are
I:bandedClimateWidth=1

# X offset for initial spawn search in blocks
I:xSpawnOffset=0

# Z offset for initial spawn search in blocks
I:zSpawnOffset=0
```

}

##########################################################################################################

# ocean control parameters

#--------------------------------------------------------------------------------------------------------#

# Frequencies are x per thousand but a little goes a very long way because seeds grow a lot. SeparateLandmasses = true makes an oceanic world with these settings andSeparateLandmasses = false makes a continental world

##########################################################################################################

"ocean control parameters" {
    # frequency of large continent seeds, about 8000x16000
    I:"Incidence of Continents,Large"=0

```
# frequency of medium continent seeds, about 4000x8000
I:"Incidence of Continents,Medium"=60

# frequency of small continent seeds, about 2000x4000
I:"Incidence of Continents,Small"=120

# frequency of large island seeds, about 500x1000
I:"Incidence of Islands,Large"=60

# frequency of medium island seeds, about 250x500, but they tend to break up into archipelagos
I:"Incidence of Islands,Medium"=30

# Rounds of continent and large island expansion in oceanic worlds (with separateLandmasses off). More makes continents larger and oceans narrower. Default is 1.
I:"Land Expansion Rounds"=3

# True mostly stops landmasses merging.With default settings you will get an oceanic world if true and a continental world if false
B:SeparateLandmasses=true
```

}

##########################################################################################################

# oceanbiomeclimates

#--------------------------------------------------------------------------------------------------------#

# Climate Types are: SNOWY,COOL,WARM,HOT,OCEAN,DEEP_OCEANMEDIUM,PLAINS,LAND, and SEA.  MEDIUM is COOL and WARM, PLAINS is COOL and WARM and HOT, LAND is all four land, SEA is both oceans

##########################################################################################################

oceanbiomeclimates {
    S:"DeepOcean climate"=DEEP_OCEAN
    S:"Ocean climate"=OCEAN
}

oceanbiomeids {
    I:"DeepOcean ID"=24
    I:"Ocean ID"=0
}

oceanbiomeincidences {
    I:"DeepOcean Incidence"=100
    I:"Ocean Incidence"=100
}

oceanbiomevillages {
    B:"DeepOcean hasVillages"=false
    B:"Ocean hasVillages"=false
}

##########################################################################################################

# vanillabiomeclimates

#--------------------------------------------------------------------------------------------------------#

# Climate Types are: SNOWY,COOL,WARM,HOT,OCEAN,DEEP_OCEANMEDIUM,PLAINS,LAND, and SEA.  MEDIUM is COOL and WARM, PLAINS is COOL and WARM and HOT, LAND is all four land, SEA is both oceans

##########################################################################################################

vanillabiomeclimates {
    S:"Birch Forest Hills M climate"=DEFAULT
    S:"Birch Forest Hills climate"=DEFAULT
    S:"Birch Forest M climate"=DEFAULT
    S:"Birch Forest climate"=Warm
    S:"Cold Taiga Hills climate"=DEFAULT
    S:"Cold Taiga M climate"=DEFAULT
    S:"Cold Taiga climate"=Snowy
    S:"Desert Hills climate"=DEFAULT
    S:"Desert M climate"=DEFAULT
    S:"Desert climate"=Hot
    S:"Extreme Hills M climate"=DEFAULT
    S:"Extreme Hills climate"=MEDIUM
    S:"Extreme Hills+ M climate"=DEFAULT
    S:"Extreme Hills+ climate"=DEFAULT
    S:"Flower Forest climate"=DEFAULT
    S:"Forest Hills climate"=DEFAULT
    S:"Forest climate"=MEDIUM
    S:"Ice Mountains climate"=DEFAULT
    S:"Ice Plains Spikes climate"=DEFAULT
    S:"Ice Plains climate"=Snowy
    S:"Jungle Hills climate"=DEFAULT
    S:"Jungle M climate"=DEFAULT
    S:"Jungle climate"=Warm
    S:"Mega Spruce Taiga climate"=DEFAULT
    S:"Mega Taiga Hills climate"=DEFAULT
    S:"Mega Taiga climate"=Cool
    S:"Mesa (Bryce) climate"=DEFAULT
    S:"Mesa Plateau F M climate"=DEFAULT
    S:"Mesa Plateau F climate"=Hot
    S:"Mesa Plateau M climate"=DEFAULT
    S:"Mesa Plateau climate"=Hot
    S:"Mesa climate"=DEFAULT
    S:"Mushroom Island climate"=DEFAULT
    S:"Plains climate"=PLAINS
    S:"Roofed Forest M climate"=DEFAULT
    S:"Roofed Forest climate"=Cool
    S:"Savanna M climate"=DEFAULT
    S:"Savanna Plateau M climate"=DEFAULT
    S:"Savanna Plateau climate"=DEFAULT
    S:"Savanna climate"=Hot
    S:"Sunflower Plains climate"=DEFAULT
    S:"Swampland M climate"=DEFAULT
    S:"Swampland climate"=Warm
    S:"Taiga (snowless) M climate"=DEFAULT
    S:"Taiga (snowless) climate"=Cool
    S:"Taiga Hills climate"=DEFAULT
}

vanillabiomeids {
    I:"Birch Forest Hills ID"=28
    I:"Birch Forest Hills M ID"=156
    I:"Birch Forest ID"=27
    I:"Birch Forest M ID"=155
    I:"Cold Taiga Hills ID"=31
    I:"Cold Taiga ID"=30
    I:"Cold Taiga M ID"=158
    I:"Desert Hills ID"=17
    I:"Desert ID"=2
    I:"Desert M ID"=130
    I:"Extreme Hills ID"=3
    I:"Extreme Hills M ID"=131
    I:"Extreme Hills+ ID"=34
    I:"Extreme Hills+ M ID"=162
    I:"Flower Forest ID"=132
    I:"Forest Hills ID"=18
    I:"Forest ID"=4
    I:"Ice Mountains ID"=13
    I:"Ice Plains ID"=12
    I:"Ice Plains Spikes ID"=140
    I:"Jungle Hills ID"=22
    I:"Jungle ID"=21
    I:"Jungle M ID"=149
    I:"Mega Spruce Taiga ID"=160
    I:"Mega Taiga Hills ID"=33
    I:"Mega Taiga ID"=32
    I:"Mesa (Bryce) ID"=165
    I:"Mesa ID"=37
    I:"Mesa Plateau F ID"=38
    I:"Mesa Plateau F M ID"=166
    I:"Mesa Plateau ID"=39
    I:"Mesa Plateau M ID"=167
    I:"Mushroom Island ID"=14
    I:"Plains ID"=1
    I:"Roofed Forest ID"=29
    I:"Roofed Forest M ID"=157
    I:"Savanna ID"=35
    I:"Savanna M ID"=163
    I:"Savanna Plateau ID"=36
    I:"Savanna Plateau M ID"=164
    I:"Sunflower Plains ID"=129
    I:"Swampland ID"=6
    I:"Swampland M ID"=134
    I:"Taiga (snowless) ID"=5
    I:"Taiga (snowless) M ID"=133
    I:"Taiga Hills ID"=19
}

vanillabiomeincidences {
    I:"Birch Forest Incidence"=10
    I:"Cold Taiga Incidence"=10
    I:"Desert Incidence"=30
    I:"Extreme Hills Incidence"=20
    I:"Forest Incidence"=20
    I:"Ice Plains Incidence"=30
    I:"Jungle Incidence"=5
    I:"Mega Taiga Incidence"=5
    I:"Mesa Plateau F Incidence"=4
    I:"Mesa Plateau Incidence"=1
    I:"Plains Incidence"=30
    I:"Roofed Forest Incidence"=10
    I:"Savanna Incidence"=20
    I:"Swampland Incidence"=10
    I:"Taiga (snowless) Incidence"=10
}

vanillabiomevillages {
    B:"Birch Forest Hills M hasVillages"=false
    B:"Birch Forest Hills hasVillages"=false
    B:"Birch Forest M hasVillages"=false
    B:"Birch Forest hasVillages"=false
    B:"Cold Taiga Hills hasVillages"=false
    B:"Cold Taiga M hasVillages"=false
    B:"Cold Taiga hasVillages"=false
    B:"Desert Hills hasVillages"=false
    B:"Desert M hasVillages"=false
    B:"Desert hasVillages"=true
    B:"Extreme Hills M hasVillages"=false
    B:"Extreme Hills hasVillages"=false
    B:"Extreme Hills+ M hasVillages"=false
    B:"Extreme Hills+ hasVillages"=false
    B:"Flower Forest hasVillages"=false
    B:"Forest Hills hasVillages"=false
    B:"Forest hasVillages"=false
    B:"Ice Mountains hasVillages"=false
    B:"Ice Plains Spikes hasVillages"=false
    B:"Ice Plains hasVillages"=false
    B:"Jungle Hills hasVillages"=false
    B:"Jungle M hasVillages"=false
    B:"Jungle hasVillages"=false
    B:"Mega Spruce Taiga hasVillages"=false
    B:"Mega Taiga Hills hasVillages"=false
    B:"Mega Taiga hasVillages"=false
    B:"Mesa (Bryce) hasVillages"=false
    B:"Mesa Plateau F M hasVillages"=false
    B:"Mesa Plateau F hasVillages"=false
    B:"Mesa Plateau M hasVillages"=false
    B:"Mesa Plateau hasVillages"=false
    B:"Mesa hasVillages"=false
    B:"Mushroom Island hasVillages"=false
    B:"Plains hasVillages"=true
    B:"Roofed Forest M hasVillages"=false
    B:"Roofed Forest hasVillages"=false
    B:"Savanna M hasVillages"=false
    B:"Savanna Plateau M hasVillages"=false
    B:"Savanna Plateau hasVillages"=false
    B:"Savanna hasVillages"=true
    B:"Sunflower Plains hasVillages"=false
    B:"Swampland M hasVillages"=false
    B:"Swampland hasVillages"=false
    B:"Taiga (snowless) M hasVillages"=false
    B:"Taiga (snowless) hasVillages"=false
    B:"Taiga Hills hasVillages"=false
}</details>

**Comment by Quizer9O8** [comment]
Oh btw tp to this cords:  X -2518 Z 2345

**Comment by Quizer9O8** [comment]
this issue is still occuring after the latest update.
It looks like the beach biome tries to become a full biome thus leaving a big rock clump behind with straight unnatural borders.

**Comment by srs-bsns** [comment]
This is a biome placement issue, and RTG only generates terrain. This is likely caused by another mod that is affecting the placement of the biomes (possibly GeographiCraft). This sort of thing has not been observed with RTG alone.

**Comment by Quizer9O8** [comment]
I really have to disappoint you there.
You see this is happening in RTG also with mod or without it.

**Comment by whichonespink44** [comment]
From Zeno:

> The problem is that the beach biome occasionally generates hills which are too high to be "beach". They can also sometimes be partly in the adjacent ocean. These are supposed to be repaired to a land biome, but if there's not one close enough - which happens on protruding beach penisulas - it fails.  If we change the beach terrain we get chunk walls so that's being put off until the next breaking change.
We might be able to patch it by having ocean above-sea-level surfacing be the same as beach.

I'll try the beach surfacing patch that Zeno suggested to see if that helps.

**Comment by srs-bsns** [comment]
This is from a few days ago using a dev build from ab6402f2035d14d2e88ab65b71cb2da2af52bd83

![2017-01-29_23 26 48](https://cloud.githubusercontent.com/assets/8453133/22543565/109a115a-e8ff-11e6-8d11-595df4b82f37.png)


What more needs to be done here?

**Comment by whichonespink44** [comment]
Is this still an issue in the latest 4.2.0.0 snapshots?

**Comment by whichonespink44** [comment]
![](https://cdn.discordapp.com/attachments/168326116761665536/331795348639186944/2017-07-03_21.18.04.png)

RTG 4.1.2.4 + BOP + GC
https://cdn.discordapp.com/attachments/168326116761665536/331807312085909514/GeographiCraft.rar

**Comment by whichonespink44** [comment]
The surfacing issue (exposed stone) has been fixed in c3049f6. The chunky biome borders may still happen rarely, but at least the surface will be sand instead of stone.
